### PR TITLE
Integrate github-runner-manager into this repo

### PR DESCRIPTION
**Pull request by javierdelapuente**
Applicable spec: <link>

### Overview

This PR copies all the files from the repo https://github.com/canonical/github-runner-manager into the current repo.

As this two repos are very close and the github-runner-manager works only with the github-runner-operator, putting the code of both repos in the same re[p can be considered a good practice, as we can simplify the dependency management, facilitate the review process and  improve the testing.

The github-runner-manager is included in a separated directory, as the goal for following refactorings is to run that code outside of the charm code, and probably it can be structured in a different wat.

The commit used from github-runner-manager is cfb8f74b93d6f29acce36cfc87ccab6d7c19a344, and the repo [github-runner-manager](https://github.com/canonical/github-runner-manager) will be archived when this PR is merged.

There may be some polishing after this PR with renovate. It will be done after some dependencies are removed in following PRs.

### Module Changes

The charm changes in the `requirements.txt` a dependency from the repo github-runner-manager to the directory github-runner-manager, where it will install the Python package.

### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied.
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied.
- [x] The changes are compliant with [ISD054 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation is generated using `src-docs`.
- [x] The documentation for charmhub is updated.
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`).
- [x] The changelog is updated with changes that affects the users of the charm.
- [x] The changes do not introduce any regression in code or tests related to LXD runner mode.

### CODE REVIEW

File: github-runner-manager/requirements.txt
Comment (cbartz) [inline github-runner-manager/requirements.txt:1]: Do you know if we have to adapt the renovate config somehow to be able to update this `requirements.txt`?

**Review by cbartz — COMMENTED** [review]
Thanks! Seems like the easiest way to integrate it back.

I think it might be worth adding a few lines to the README explaining that this repo contains the code for the charm and the application, and that developers need to explicitly enable the tox envs when coding.

File: .github/workflows/test_github_runner_manager.yaml
[the remark this answers is not in the document]
Comment (javierdelapuente) [reply .github/workflows/test_github_runner_manager.yaml:1]: yes, I will add it to is-charms-github-repos once this PR is merged

File: github-runner-manager/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+source venv/bin/activate
+```
+
+## Generating src docs for every commit
+
+Run the following command:
+
+```bash
+echo -e "tox -e src-docs\ngit add src-docs\n" >> .git/hooks/pre-commit
+chmod +x .git/hooks/pre-commit
```
[the remark this answers is not in the document]
Comment (javierdelapuente) [reply github-runner-manager/CONTRIBUTING.md:16]: There are several ways of doing this, like doing it in the root `tox -e src-docs`. I will add it just to the CONTRIBUTING.md instructions to run both `tox` files.

File: github-runner-manager/requirements.txt
Comment (javierdelapuente) [reply github-runner-manager/requirements.txt:1]: I think so

**Comment by yhaliaw** [comment]
How should the code be organized?

I was thinking of top-level directories like the following:
- github-runner-manager
- github-runner-operator

Which seems like a smaller change and allows multiple different charms in one repo if needed.
What are the advantage to the approach taken in the PR?

**Comment by javierdelapuente** [comment]
> How should the code be organized?
> 
> I was thinking of top-level directories like the following:
> 
> * github-runner-manager
> * github-runner-operator
> 
> Which seems like a smaller change and allows multiple different charms in one repo if needed. What are the advantage to the approach taken in the PR?

The main reason I went this way is simplicity. In the file `requirements.txt` for the charm, I cannot reference a directory outside the current directory (so we would have . Besides, the `github-runner-manager` is something that we will only use in the charm/application of the `github-runner`.

If at some point we decide to add another charm, I would just refactor at that point.

**Review by cbartz — APPROVED** [review]

**Review by yhaliaw — APPROVED** [review]

**Pull request merged**